#! /bin/sh /usr/share/dpatch/dpatch-run
## 73_list-id_strict_rfc.dpatch by  <lionel@mamane.lu>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Make sure the List-Id header strictly conforms to RFC 2919:
## DP: always put the list-id between brackets.

@DPATCH@
diff -urNad mailman-2.1.5~/Mailman/Handlers/CookHeaders.py mailman-2.1.5/Mailman/Handlers/CookHeaders.py
--- mailman-2.1.5~/Mailman/Handlers/CookHeaders.py	2003-12-18 15:58:22.000000000 +0100
+++ mailman-2.1.5/Mailman/Handlers/CookHeaders.py	2005-11-13 18:14:47.546324750 +0100
@@ -172,7 +172,10 @@
         desc = str(h)
     else:
         desc = ''
-    listid_h = formataddr((desc, listid))
+    if (desc == ''):
+        listid_h = '<%s>' % listid
+    else:
+        listid_h = formataddr((desc, listid))
     # BAW: I think the message object should handle any necessary wrapping.
     del msg['list-id']
     msg['List-Id'] = listid_h
